### ISSUE TITLE
steemit market change your prices setting a bug that must be fixed immediately 

### ISSUE THREAD

**Issue by TheGoldenPhoenix** [steemit/condenser#2164]
## The Reported bug is related to steemit.com/market:
For steemit traders or maybe just simple users who want to exchange their liquid sbd/steem this is a major problem.
 the problem is when you **set your price** in https://steemit.com/market and **place your order** you will see that the prices has changed this can cause a big lose or a win in some cases to the trader and this must be fixed soon.
#### Screenshot 1 (**Setting the price of sbd/steem  to buy/sell  0.185/1.05**)
![image](https://res.cloudinary.com/hpiynhbhq/image/upload/v1513292154/onnba8dv5bdh1qnhoh9o.png)

after placing the order this is the prices that we get
#### Screenshot 2 (**the prices after placing the order**)
![image](https://res.cloudinary.com/hpiynhbhq/image/upload/v1513292362/fcayaybuajmgtcwx7x3o.png)

to confirm that i didn t change the prices here is another screenshot 
#### Screenshot 3 (**confirmation **)
![image](https://res.cloudinary.com/hpiynhbhq/image/upload/v1513292533/aane6lethywih0ohglyq.png)
![image](https://res.cloudinary.com/hpiynhbhq/image/upload/v1513292693/kgcwnwlyxvkpq3mbvmvj.png)
#### Gif  (**confirmation **) of the issue
<centre> https://media.giphy.com/media/3o6nUVqrn5Ug5lpo76/giphy.gif </centre>

## The Testing Environment Details
Desktop : https://steemit.com/market
OS: Windows 10
Browser: Google Chrome Version 63.0.3239.108 (Official Build) (64-bit)
![image](https://res.cloudinary.com/hpiynhbhq/image/upload/v1513293348/rchiz9jc0774byor8xzv.png)

## Steps to reproduce the bug :
1- Open **https://steemit.com/market** on your browser.
2- Login using your master key or password.
3- Now try if you have some liquid steem/sbd in your wallet try make an order and see your self.
![image](https://res.cloudinary.com/hpiynhbhq/image/upload/v1513293574/ykkulqztbyybhhb9ggej.png)
**Note**: i used small amount of sbd/steem because this is what i have in my wallet now ;p
thank you for reading & have a good day (^_^)

<hr>
<centre><h1><blockquote>𝕨𝕖 𝕘𝕠 𝕤𝕝𝕠𝕨 𝕓𝕦𝕥 𝕨𝕖 𝕘𝕠 𝕗𝕦𝕣𝕥𝕙𝕖𝕣 @thegoldenphoenix</blockquote></centre>
<hr>

<br /><hr/><em>Posted on <a href="https://utopian.io/utopian-io/@thegoldenphoenix/steemit-market-change-your-prices-setting-a-bug-that-must-be-fixed-immediately">Utopian.io -  Rewarding Open Source Contributors</a></em><hr/>

**Comment by mvandeberg** [comment]
There are always going to be some rounding errors for small trades.

Trading 0.001 STEEM for 1.05 SBD would result in 0.00105 SBD, but SBD only has a precision of 3. The only trades that are available are a price of 1.00 SBD/STEEM or 2.00 SBD/STEEM.

If you assume the price of 0.185 as truth for 0.5 STEEM, the order would be for 0.0925 SBD. Again, SBD does not have the precision for this amount. So you either need to trade 0.093 SBD or 0.092 SBD. But these amounts now result in a price of 0.186 SBD/STEEM or 0.184 SBD/STEEM respectively.

TLDR: Trading with high precision STEEM/SBD results in a lower precision price.

**Comment by roadscape** [comment]
Maybe we should consider imposing a minimum order size on the frontend? Dust orders seem to cause more issues than they're worth.

Also, I believe the market page at one point used to highlight the price field when a precision issue was detected.

**Comment by TheGoldenPhoenix** [comment]
mvandeberg i think that you missed the note 
![image](https://user-images.githubusercontent.com/34288299/34127649-33075a82-e43e-11e7-9119-971cf084797a.png)
and for the precision this plays a big role in trading specially when you are facing a trading bot you must be smart to win the game ;)

**Comment by mvandeberg** [comment]
No, I noticed that.

There are two things here.

1. Understanding how the blockchain level protocol works. In this case it is understanding that the prices you want are not expressible with the given quantities.

2. Communicating that reality to the user in the UI.

**Comment by TheGoldenPhoenix** [comment]
okey let us then pretend that the prices of steem/sbd goes up >100$ what will happen ???
so instead of imposing the minimum orders size like roadscape said i think for the short term a message in the front page will fix the issue and for the long term this must be fixed

**Comment by muksihs** [comment]
IMHO: As a user who trades very small amounts, preventing me from trading because I trade small amounts is not the correct fix. There already is a dialog which pops up to confirm the "actual calculated price" when placing orders, why is this not sufficient?

**Comment by roadscape** [comment]
Hi, if this issue persists please file it on the wallet repo: https://github.com/steemit/wallet/issues
